src/helpers/getTokenFromUrl.ts

```diff
@@ -0,0 +1,17 @@
+// This is part of the authentication code used to obtain a token from Spotify during login.
+interface Token {
+  [key: string]: string;
+}
+
+export default function getTokenFromUrl() {
+  return window.location.hash
+    .substring(1)
+    .split("&")
+    .reduce((initial: Token, item: string) => {
+      console.log(initial);
+      console.log(item);
+      const parts = item.split("=");
+      initial[parts[0]] = decodeURIComponent(parts[1]);
+      return initial;
+    }, {});
```

Comment:
```suggestion
export default function getTokenFromUrl() {
  const urlHashParams = window.location.hash
    .substring(1)
    .split("&")
    .reduce((initial, item) => {
      const parts = item.split("=");
      initial[parts[0]] = decodeURIComponent(parts[1]);
      return initial;
    }, {} as Record<string, string | undefined>);
  
    return urlHashParams.access_token;
```

Reply:
The `Token` type is correct but the name is misleading. I would rather call it something like `UrlHashParams`. There's also the helper type `Record` which is basically the same, so it's a matter of taste which one to use. I decided to add it inline, in the reduce argument. It teaches you something about type inference (no need to define types for the reduce function arguments this way) but you might find it harder to read so again, a matter or taste. Additionally, I added `| undefined` to the type of the values just to make sure the rest of our code is aware that we might get `undefined` for `urlHashParams.access_token` if it's not in the URL.

Obviously, with this change, you'd have to remove `.access_token` from the `getTokenFromUrl` call in `App.tsx`.